lib/active_storage/async_uploader.rb

```diff
@@ -0,0 +1,34 @@
+require "concurrent/promise"
+
+class ActiveStorage::AsyncUploader
+  class << self
+    def result uploaders
+      promises = uploaders.map(&:promise)
+      Concurrent::Promise.zip(*promises).value!
+    end
+  end
+
+  attr_reader :promise
+
+  def initialize service, key, checksum: nil
+    @data = ""
```

Comment:
I'm not quite sure if using a string instead of an enumerator is a problem. I'm also open for a new class name ideas.